### PR TITLE
[ART-9247] Create FIPS informing job

### PR DESCRIPTION
**Pull request by ashwindasr**
Add a nightly informing job for FIPS scan

### CODE REVIEW

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

**Comment by ashwindasr** [comment]
/pj-rehearse pull-ci-openshift-release-master-nightly-4.16-fips-payload-scan

File: ci-operator/step-registry/openshift/release/fips/openshift-release-fips-commands.sh [hunk 1]
```diff
@@ -4,12 +4,56 @@ set -o nounset
...
+payload_pullspec=$(oc get clusterversion version -o json | jq -r .status.desired.image)
 
-./check-payload scan payload -V "$major_minor" --url "${RELEASE_IMAGE_LATEST}"
+#if [[ "$payload_url" == *"@sha256"* ]]; then
+#    payload_url=$(echo "$payload_url" | sed 's/@sha256.*/:latest/')
+#fi
+
+echo $KUBECONFIG
+ls $SHARED_DIR
+
```
Comment (bradmwilliams) [inline ci-operator/step-registry/openshift/release/fips/openshift-release-fips-commands.sh:41]: Is any of this is necessary?

File: ci-operator/config/openshift/release/openshift-release-master__nightly-4.16.yaml [hunk 2]
```diff
@@ -924,6 +924,15 @@ tests:
       TEST_TYPE: conformance-parallel
     workflow: openshift-e2e-powervs-ipi
   timeout: 6h0m0s
+- as: fips-payload-scan
+  steps:
```
Comment (bradmwilliams) [inline ci-operator/config/openshift/release/openshift-release-master__nightly-4.16.yaml:928]: I think you want this to be a `periodic` job...
You can add the following to enable a periodic:
```
interval: 168h
```

File: ci-operator/config/openshift/release/openshift-release-master__nightly-4.16.yaml [hunk 3]
```diff
@@ -924,6 +924,15 @@ tests:
...
+    cluster_profile: aws
+    env:
+      MAJOR_MINOR: "4.16"
+    test:
+    - ref: openshift-release-fips
+    - ref: clusterbot-wait
+    workflow: ipi-aws
```
Comment (bradmwilliams) [inline ci-operator/config/openshift/release/openshift-release-master__nightly-4.16.yaml:935]: I would go ahead and add this configuration to all the versions where the job will be ran from.
After this PR merges, you can chase it with the necessary update to add informing jobs to the respective releases.

**Review by bradmwilliams — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/openshift/release/fips/openshift-release-fips-commands.sh (hunk 1 above)
Comment (ashwindasr) [reply ci-operator/step-registry/openshift/release/fips/openshift-release-fips-commands.sh]: Apologies, cleaned

**Comment by ashwindasr** [comment]
/pj-rehearse ack

The errors are not an issue. I'll update the excludes with check-payload

File: ci-operator/step-registry/openshift/release/fips/openshift-release-fips-commands.sh [hunk 4]
```diff
@@ -4,12 +4,51 @@ set -o nounset
...
+else
+    echo "Fail to create $namespace namespace."
+    pass=false
 fi
 
-# Get major.minor from payload url
-major_minor=$(echo "${RELEASE_IMAGE_LATEST}" | awk -F':' '{print $2}' | awk -F'.' '{print $1"."$2}')
 
-./check-payload scan payload -V "$major_minor" --url "${RELEASE_IMAGE_LATEST}"
+payload_pullspec=$(oc get clusterversion version -o json | jq -r .status.desired.image)
```
Comment (stbenjam) [inline ci-operator/step-registry/openshift/release/fips/openshift-release-fips-commands.sh:34]: I think this ends up just checking the payload image on the build cluster? You probably really do want to check `${RELEASE_IMAGE_LATEST}` so you're checking the payload under test

Comment (stbenjam) [reply ci-operator/step-registry/openshift/release/fips/openshift-release-fips-commands.sh:34]: Disregard, we cleared this up on Slack

**Pull request merged**